src/Chainweb/ChainId.hs

```diff
@@ -89,18 +89,21 @@ instance Exception ChainIdException
 -- -------------------------------------------------------------------------- --
 -- ChainId
 
--- | ChainId /within a Chainweb/.
+-- | ChainId /within the context a Chainweb instance/.
 --
--- Generally a block chain is /globally/ uniquely identified by its genesis hash.
--- This type only uniquely identifies the chain /locally/ within the context of
--- a chainweb.
+-- The set of valid ChainIds is determined by the 'ChainwebVersion'. In almost
+-- all use cases there should be a context that is an instance of
+-- 'HasChainwebVersion' can be used get the set of chain ids.
 --
--- However, the chainweb context is globally uniquely identified by the
--- 'ChainwebVersion', which in turn is determined by the identities of the
--- chains in the chainweb. Since the chainweb topology is statically represented
--- on the type level, while the Chainweb version is a runtime value, we break
--- the cycle by using the Chainweb version as globally unique identifier and
--- include the 'ChainwebVersion' and the 'ChainId' into the genesis hash.
+-- In the context of a particular chain the respective 'ChainId' can be obtained
+-- via instances of 'HasChainId'.
+--
+-- /How to create values of type 'ChainId'/
+--
+-- * To fold or traverse over all chain ids, use 'chainIds'.
```

Comment:
Add a note that all of these functions are in `Chainweb.Version`.

Better would be to consider how to make a single import, since at this point you'll be importing `Chainweb.ChainId` just for the type and `Chainweb.Version` for everything useful. Consider 
- combining `Chainweb.ChainId` and `Chainweb.Version` modules
- exporting all of ChainId's API out of `Chainweb.Version`
- Renaming `Version` (kind of uninspiring name for an entire Chainweb network) to justify including the `ChainId` API. `NetworkId`or `ChainwebId` or something

Reply:
> exporting all of ChainId's API out of Chainweb.Version

Yes, that's a good idea.

> Renaming Version (kind of uninspiring name for an entire Chainweb network)

That's a longstanding issue. It was brought up during the very first Chainweb-code review, but it never was high priority. `NetworkId` is already taken by the P2P network (we would have to rename the type there first to avoid confusion). What about `ChainwebInstance`?

Changing the name would be straight forward, but would touch a lot of code. So, maybe we should do it during vacation time or some holiday when there are not a lot of open PRs?

Reply:
OK let's just do the exports for now.